From 7ac1a9ae2f917d305be20ca314ae64d19d415d0f Mon Sep 17 00:00:00 2001
From: Sebastien Guiriec <s-guiriec@ti.com>
Date: Thu, 18 Nov 2010 23:01:53 +0100
Subject: ASoC: ABE HAL: Add API to disable ABE IRQ

Add ABE HAL API to disable ABE IRQ (DMA req/MCU).

Change-Id: Ifa698c548fe9344ecb3b71a120e2f3bda9c22e1e
Signed-off-by: Sebastien Guiriec <s-guiriec@ti.com>
Signed-off-by: Misael Lopez Cruz <misael.lopez@ti.com>
Signed-off-by: Chris Kelly <c-kelly@ti.com>
---
 sound/soc/omap/abe/abe_api.h |    6 ++++++
 sound/soc/omap/abe/abe_ini.c |   19 +++++++++++++++++++
 2 files changed, 25 insertions(+), 0 deletions(-)

Index: Kernel/sound/soc/omap/abe/abe_api.h
===================================================================
--- Kernel.orig/sound/soc/omap/abe/abe_api.h	2011-01-19 00:14:35.342073087 +0530
+++ Kernel/sound/soc/omap/abe/abe_api.h	2011-01-19 00:14:39.022073720 +0530
@@ -98,6 +98,12 @@
  */
 abehal_status abe_irq_clear(void);
 /**
+ * abe_disable_clear - disable MCU/DSP ABE interrupt
+ *
+ * This subroutine is disabling ABE MCU/DSP Irq
+ */
+abehal_status abe_disable_irq(void);
+/**
  * abe_select_main_port - Select stynchronization port for Event generator.
  * @id: audio port name
  *
Index: Kernel/sound/soc/omap/abe/abe_ini.c
===================================================================
--- Kernel.orig/sound/soc/omap/abe/abe_ini.c	2011-01-19 00:14:18.522072429 +0530
+++ Kernel/sound/soc/omap/abe/abe_ini.c	2011-01-19 00:14:39.022073720 +0530
@@ -46,6 +46,25 @@
 	abe_block_copy(COPY_FROM_HOST_TO_ABE, ABE_ATC, 0x3C, &atc_reg, 4);
 }
 /**
+ * abe_disable_clear - disable MCU/DSP ABE interrupt
+ *
+ * This subroutine is disabling ABE MCU/DSP Irq
+ */
+abehal_status abe_disable_irq(void)
+{
+	u32 atc_reg;
+
+	/* disables the DMAreq from AESS AESS_DMAENABLE_CLR = 127
+	 * DMA_Req7 will still be enabled as it is used for ABE trace */
+	atc_reg = 0x7F;
+	abe_block_copy(COPY_FROM_HOST_TO_ABE, ABE_ATC, 0x64, &atc_reg, 4);
+	/* disables the MCU IRQ from AESS to Cortex A9 */
+	atc_reg = 0x01;
+	abe_block_copy(COPY_FROM_HOST_TO_ABE, ABE_ATC, 0x40, &atc_reg, 4);
+
+	return 0;
+}
+/**
  * abe_build_scheduler_table
  *
  */
